File: smite/src/main/java/hirez/smite/object/Team.java
package hirez.smite.object;

import com.fasterxml.jackson.annotation.JsonProperty;
import hirez.api.object.interfaces.ReturnedMessage;
import lombok.Data;

@Data
public class Team implements ReturnedMessage {

    @JsonProperty("Founder")
    private final String founder;
    @JsonProperty("FounderId")
    private final String founderId;
    @JsonProperty("Losses")
    private final long losses;
    @JsonProperty("Name")
    private final String name;
    @JsonProperty("Players")
    private final long players;
    @JsonProperty("Rating")
    private final long rating;
    @JsonProperty("ret_msg")
    private final String returnedMessage;
    @JsonProperty("Tag")
    private final String tag;
    @JsonProperty("TeamId")
    private final long teamId;
    @JsonProperty("Wins")
    private final long wins;

}

File: api/src/main/kotlin/hirez/json/team.kt
package hirez.json

import com.google.gson.annotations.SerializedName
import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class TeamMember(
			@SerializedName("ret_msg")
			override val returnMessage: String?,
			@SerializedName("AccountLevel")
			val level: Int,
			@SerializedName("JoinedDatetime")
			val createdAt: Date,
			@SerializedName("LastLoginDatetime")
			val updatedAt: Date,
			@SerializedName("Name")
			val name: String
) : ReturnMessage

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Team(
			@SerializedName("ret_msg")
			override val returnMessage: String,
			val founder: String,
			val founderId: Long,
			val losses: Int,
			val name: String,
			val players: Int,
			val rating: Int,
			val tag: String,
			val teamId: Long,
			val wins: Int
) : ReturnMessage